noir-projects/aztec-nr/aztec/src/macros/notes/mod.nr

```diff
@@ -69,28 +65,98 @@ comptime fn generate_note_interface(s: StructDefinition, note_type_id: Field) ->
             fn get_note_type_id() -> Field {
                 $note_type_id
             }
+        }
+    }
+}
+
+/// Generates default `NoteHashing` trait implementation for a given note struct `s` and returns it as a quote.
+///
+/// # Generated Implementation
+/// ```
+/// impl NoteHashing for NoteStruct {
+///     fn compute_note_hash(self, storage_slot: Field) -> Field { ... }
+///
+///     fn compute_nullifier(self, context: &mut PrivateContext, note_hash_for_nullify: Field) -> Field { ... }
+///
+///     unconstrained fn compute_nullifier_without_context(
+///         self,
+///         storage_slot: Field,
+///         contract_address: AztecAddress,
+///         note_nonce: Field,
+///     ) -> Field { ... }
+/// }
+/// ```
+comptime fn generate_note_hashing(s: StructDefinition) -> Quoted {
+    let name = s.name();
 
+    quote {
+        impl aztec::note::note_interface::NoteHashing for $name {
             fn compute_note_hash(self, storage_slot: Field) -> Field {
                 let inputs = aztec::protocol_types::utils::arrays::array_concat(self.pack(), [storage_slot]);
                 aztec::protocol_types::hash::poseidon2_hash_with_separator(inputs, aztec::protocol_types::constants::GENERATOR_INDEX__NOTE_HASH)
             }
+
+            fn compute_nullifier(
+                self,
+                context: &mut aztec::prelude::PrivateContext,
+                note_hash_for_nullify: Field,
+            ) -> Field {
+                let owner_npk_m = aztec::keys::getters::get_public_keys(self.owner).npk_m;
```

Comment:
!!!

This is no bueno, you're blindly assuming that the note will have an `owner` field. We can maybe think of a way to autoderive this basic nullifier, but we should not do it this way.

Reply:
and we're also assuming `AztecAddress<owner>` does have PublicKeys associated, else the error should look like "No public key registered for address ... Register it by calling pxe.registerAccount(...)"

Reply:
i think we should re-discuss the concept of "owner" and if it belongs here, or a more-formal definition should be done and expectable from such a field.

if we choose "nelly" for example, we can expect both (1) it exists on NoteHashing and (2) it has PublicKeys

but "owner" can be related to "who has as msg_sender authorization to do X", which doesn't reflect exactly in this kind of "ownership" that NoteHashing applies, specially (2) having PublicKeys

either we move "owner" to "nelly"
or we move "owner (bc of being msg_sender)" to "admin" terminology, and we assume that "owner" is "he who nullifies notes" and NEEDS to be expected here of course

Reply:
As discussed in DMs inserted `assert_has_owner` check into the note macros that checks owner is present. This should be changed in the future. If a dev uses `#[custom_note]` macro the check is not applied as the dev needs to implement the `NoteHash` interface on his own.